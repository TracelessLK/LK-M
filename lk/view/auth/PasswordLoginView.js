
import React, { Component } from 'react'
import {

  Text,
  TextInput,
  View
} from 'react-native'
import {
  Card,
  CardItem,
  Button,
  Toast
} from 'native-base'
const md5 = require('crypto-js/md5')
const {engine} = require('LK-C')

let Application = engine.getApplication()
const lkApplication = Application.getCurrentApp()
const style = require('../style')

export default class PasswordLoginView extends Component<{}> {
  static navigationOptions =() => {
    return {
      headerTitle: '登录'
    }
  }
  constructor (props) {
    super(props)
    const {user} = this.props.navigation.state.params
    this.user = user
    this.state = {}
    this.t = ''
  }

  login = () => {
    const {navigation} = this.props

    // console.log({t: this.t})
    if (!this.t) {
      Toast.show({
        text: '请输入密码',
        position: 'top'
      })
    } else {
      const password = md5(this.t).toString()
      if (password === this.user.password) {
        lkApplication.setCurrentUser(this.user)
        navigation.navigate('MainStack')
      } else {
        Toast.show({
          text: '密码错误,请重试',
          position: 'top'
        })
      }
    }
  }

  render () {
    // console.log(style.mainColor)
    return (
      <View style={{alignItems: 'center', marginVertical: 40}}>
        <Card style={{width: '90%'}}>
          <CardItem header bordered>
            <Text>请输入登录密码</Text>
          </CardItem>
          <CardItem bordered>
            <View style={{justifyContent: 'center', flexDirection: 'row', alignItems: 'center'}}>
              <TextInput style={{
                flex: 1,
                color: 'black',
                fontSize: 16,
                paddingHorizontal: 4,
                borderWidth: 1,
                borderColor: '#d0d0d0',
                borderRadius: 5,
                marginHorizontal: 5,
                minHeight: 40,
                backgroundColor: '#f0f0f0',
                marginBottom: 5
              }} autoFocus secureTextEntry onChangeText={t => {
                this.t = t
              }} onSubmitEditing={this.login}
              ></TextInput>
              <Button block style={{width: 60, height: 40, backgroundColor: style.color.mainColor}} onPress={this.login}>
                <Text style={{color: 'white'}}>确认</Text>
              </Button>
            </View>

          </CardItem>
        </Card>
      </View>
    )
  }
}

PasswordLoginView.defaultProps = {

}

PasswordLoginView.propTypes = {

}
