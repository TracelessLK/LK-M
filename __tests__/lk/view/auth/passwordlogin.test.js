import React from 'react'
import 'react-native'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import PasswordLoginView from '../../../../lk/view/auth/PasswordLoginView'

jest.mock('../../../../lk/view/auth/PasswordLoginView')

describe('passwordloginview test', () => {
  let component
  let textInput
  const defaultState = {text: ''}

  test('renders correctl', () => {
    const password = renderer.create(<PasswordLoginView />).toJSON()
    expect(password).toMatchSnapshot()
  })

  beforeEach(() => {
    component = shallow(<PasswordLoginView />)
    textInput = component.find('TextInput')
  })

  it('has default state', () => {
    expect(component.state()).toEqual(defaultState)
  })

  it('renders welcome text', () => {
    const expectedText = '请输入登录密码'
    const text = component.find('Text').children().text()
    expect(text).toEqual(expectedText)
  })

  it('password is null', () => {
    expect(component.login()).toEqual(defaultState)
  })

  it('renders input field with placeholder', () => {
    const expectedPlaceholder = 'write something'
    expect(textInput.length).toBe(0)
    expect(textInput.props().placeholder).toEqual(expectedPlaceholder)
  })

  describe('when text changes', () => {
    const newTextValue = 'random string'
    beforeEach(() => {
      textInput.simulate('changeText', newTextValue)
    })

    it('updates component state', () => {
      expect(component.state().text).toEqual(newTextValue)
    })
  })
})
