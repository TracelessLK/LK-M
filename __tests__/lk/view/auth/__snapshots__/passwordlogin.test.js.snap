// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passwordloginview test renders correctl 1`] = `null`;
